### DISCUSSION TITLE
Setting up HyperHDR and the Grabber

### DISCUSSION THREAD

**Discussion by D3ltorohd** [awawa-dev/HyperHDR#644] [Q&A]
Hello Com,

today i have start an LXC and there i install HyperHDR, passthrough the USB Video Grabber MS2109 USB3.0
In Hyper i select them, and i have a live video feed. So i think it works.

What i have to set up in the config ?

Hardware Setup FireTV Cube 4k @60Hz -> Splitter Out HDMI 2 downscale to 1920X1080 @ 60Hz to the Grabber.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/1a9177af-4dc0-4e3c-b9f5-8c96ac42ff8e)

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/62fe91af-aa90-457c-9b71-1d2b3676cad7)

Greetings,
D3ltoro

**Comment by awawa-dev** [comment]
Hi
Ms2109 is not USB3.0 capable video grabber, just on old generation USB2.0 device beneath. You should choose possible lowest resolution/highest frame rate as per wiki: https://github.com/awawa-dev/HyperHDR/wiki/Quick-start
Usually yuv/nv12 should be faster although it's not always the case for USB2.0 grabbers, when somethimes mjpeg could take a lead because of smaller data  bandwidth. Verify your statistics on the HyperHDR main page.

**Comment by D3ltorohd** [comment]
You mean this.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/28d9eac4-5a5c-4430-99e6-a4c072da8855)

Something like this ?

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/7aa9efb5-9bb4-4865-b9c4-1915c189fa91)

**Comment by awawa-dev** [comment]
The grabber rejected your 640x480x60 yuyv  and fallback to 640x480 and only 10 FPS which is definitely not good (and despite what it claims it is not USB3.0 grabber, it's just a label, lsusb should tell the truth). You should not select the video mode manually because not all of combinations are allowed. Use the yellow 'info' button on the grabber page (your first screenshot) and select the mode using resolution/codec and framerate buttons (it will filter out forbidden/not supported combination).

**Comment by D3ltorohd** [comment]
So thats the best solution, when i understand you right.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/614c0d49-5fba-41da-bfcf-0f8dd2e0966f)

with yuv i have only 30 fps. i will try this settings from the screen, and will see how good it will be.

**Reply by awawa-dev** [reply]
Yes, 60fps is not available for yuv because USB2.0 is limiting the bandwidth so only compressed format like mjpeg
 can handle higher framerate and nv12 is not available for Linux. It's the same grabber: https://www.hyperhdr.eu/2020/11/grabber-for-hyperhdr-rullz-usb-20.html If you want high framerate for yuv and bit lower latency overall, you need USB3.0 capable ms2130: https://github.com/awawa-dev/HyperHDR/discussions/499

**Comment by D3ltorohd** [comment]
So they told me shit, that this is 3.0. But with lsusb he tells me usb 3.0 to
Bus 001 Device 015: ID 534d:2109 MacroSilicon USB3. 0 capture
On the web site, usb 3.0 to

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/10584746-dc1d-43e6-a2fc-13179dc314c6)

Is the ms2109 good for ambilight or should i buy the new model ?

**Reply by awawa-dev** [reply]
Unfortunately. First, they started selling such counterfeit ms2109 as usb3.0 on Ali, then it became almost "standard" everywhere. But the diagram doesn't lie: it's still USB 2.0, even if you attach a blue USB3.0 plug to it.
ms2109 (usb2.0) https://lin-ker.com/ms2109.html
ms2130 (usb3.0) https://lin-ker.com/ms2130.html

**Reply by awawa-dev** [reply]
Give ms2109 a chance because the video quality is almost as good as ms2130 (and this ms2130 firmware is still evolving, there are some minor issues) and much better than most other grabbers on the market.

**Comment by D3ltorohd** [comment]
So maybe i should use yuv and 30 fps, better than mpeg and 60 fps ?

**Reply by awawa-dev** [reply]
Test both and verify the performance stats on the main page. yuv should have a better quality and maybe 30fps will be enough (also depends on the video source: movies vs games). You can also reduce delay in the smoothing (or turn it off completely) when you are using yuv at such low framerate.

**Comment by D3ltorohd** [comment]
Input is FireTV Cube 4k @ 60Hz at the tv. I think 60 hz and 30hz fit, because 30 hz can go 2x in 60 hz.
Tomorrow i put the led's to the back of my tv. I have a test run yesterday and the leds seems to be working with hyperhdr.

**Reply by awawa-dev** [reply]
Video source hardware refresh rate is one thing, but most of the movies are around 24Hz, unless they are uprated at the source instead of TV. TV upscaler/montion compensation unit is usually quite capable so I prefer it to do its job.

**Comment by D3ltorohd** [comment]
So i have done mi setup. It works nice. Small changes i think, but so it works good. I have switched to feintech splitter, because the ezcoo take my tv screen into black and audio off, for some seconds, and this often. And CEC did not work. With feintech it works very good.

Now i have to see, how i can put hyperhdr into standby over the day, when i did not need ambilight. I have to search for a way that i can use my iobroker to send the json code to hyper, for put the instance off, that was standby right ? So he did not need to handle the video from the grabber when not needed.

**Comment by D3ltorohd** [comment]
And i have to put the lut into hyper hdr vor the ms2109. Can it run always on hdr ? Or have to switch it manual between hdr und sdr ?

**Comment by awawa-dev** [comment]
1 automatic signal detection: https://github.com/awawa-dev/HyperHDR/issues/93 to save resources assuming the grabber returns static (e.g. no signal) board after disconnecting the video configuration.
2 sdr/hdr can be turned on or off permanently, changed remotely via Json/GET-link/MQTT, or changed via TV remote control if CEC is working properly: if a pass-through is available, it usually means the Rpi is connected directly to one of the TV's HDMI sockets, even if it is not used because there is another main path using an HDMI grabber/splitter.

**Comment by D3ltorohd** [comment]
1. I dont understand. When the tv runs on day, the grabber and splitter runs too. And so HyperHDR become data from the USB Grabber and work with this right ? But the led' are off because the socket are not on. They goes on at Evening when dark outside. So in the Daytime HyperHDR should go to sleep or deactivate the instanze so he did not have to work.
2. How can i control HyperHDR through mqtt ? Can i shut an instance off with mqtt ?
3. Or was there a way to detect, when the wemos or so is not online, hyperhdr paused his work or switch the instance with the wemos how is offline, off.

**Comment by D3ltorohd** [comment]
Ok i have connected hyperhdr with my mqtt broker, there was an dp with for json api, but it will not work. I let create the link or the json from hyperhdr, but both did not work.

[http://192.168.178.155:8090/json-rpc?request=%7B"command":"instance","subcommand":"startInstance","instance":0%7D](url)

`{
    "command": "instance-startInstance",
    "error": "Can't start HyperHDR instance index 0",
    "success": false,
    "tan": 0
}`

Instance index 0 is my first instance right ? I have only on instance for now.

**Reply by awawa-dev** [reply]
You cant stop 0 instance but you can disable and later enable all instances components. Different command (first in the HyperHDR json API page).

**Comment by D3ltorohd** [comment]
I tried this to, but the link will not work. Also the json in the dp from mqtt.

EDIT:: Oh this works, when i stop all instances.

`{
	"command":"componentstate",
	"componentstate":
	{
		"component":"ALL",
		"state": false
	}
}`

So the first instance cant be stopp alone. Only the next instance like 1,2,3 and so on can be stopped seperat.

So this methode "stop all instances" are ok, to let hyperhdr sleep and he did not work. So this i can build into my javascript, so it runs only when i ambilight need !!

To the LUT Table, where i have to put it into, when i download the LUT File ? I cant find nothing to upload it in the web ui !

**Reply by awawa-dev** [reply]
It works for me and other. Provide the command/link you use. Also verify logs to see what's happening. It should disable all LEDs and the grabber at once.

**Reply by awawa-dev** [reply]
If you are using the latest v20 beta then use the installer (Advanced->LUT->first menu item), HyperHDR will automatically download and install the selected LUT version. For older versions like v19, you need to download, unpack LUT archive yourself and upload it to your HyperHDR home directory (the path is in the first lines of the HyperHDR logs right after startup). LUT is necessary for correct functioning both SDR and HDR and does not need to be deleted if you are using SDR only (uncheck only HDR tone mapping in the grabber configuration).

**Comment by D3ltorohd** [comment]
I cant update over the ui ? I have to update over command line with apt upgrade right ?

**Reply by awawa-dev** [reply]
Yes, you must manually install the package using apt install after uninstalling the old version (details can be found in the wiki). The current beta installers can be found in the latest Github Action of the master branch. Currently located here: https://github.com/awawa-dev/HyperHDR/actions/runs/6316763671 artifacts archives at the bottom of the page.

**Comment by D3ltorohd** [comment]
Hm after uninstall and install the newest version. Hyper did not start, or tells me IP Refused the connection.

Maybe there is a config where is local ip. I have this problem on other systems to, that i have to change the ip 127.0.0.0.1 to 0.0.0.0.0 that other pc in my network can reach the web ui.

When i use this command :: `/usr/bin/hyperhdr -d` it works. I can reach the web ui.

That was in the command window.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/427cc83a-5c10-4b86-8cdd-9d660383dd88)

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/a4c22775-9039-4107-9d1c-56cb0c1f14df)

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/992062b2-f4e3-47d1-af48-b0cf3d4b58b8)

The last one looks good. And i can enter the webui. But when i restart the system i cant. Only with `/usr/bin/hyperhdr -d` it works

How can i start or stop the service on linux bookworm ? The service for pi did not work.

I have change the systemd script like this. I changed the user pi to my user, and now i can start hyperhdr.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/9f4a2c07-4f53-4c4f-8196-374e4dcfb3a3)

Maybe a issue in v20 ? With v19 it runs after install.

But after restart i have to start it manual over systemctl. So i have to manual create an autostart, when the system starts ?

**Comment by awawa-dev** [comment]
You need to pay attention to what is in the output when the HyperHDR deb package is installed with apt. There should be information whether the HyperHDR service is installed (the service is disabled if a GUI is detected and the application should be launched from the menu or autostart) and what user it uses. In general, you should avoid all edits to the service file. It takes the user from the systemd service as a parameter and does not need any changes: take a look, it's %i so different than your version https://github.com/awawa-dev/HyperHDR/blob/master/cmake/debian/service/hyperhdr.systemd . Also a conflict may occur if you have another, custom copy of this service file.

**Comment by D3ltorohd** [comment]
Without editing, he will start with user pi. I have no user pi. It runs on Debian Bookworm.

**Comment by awawa-dev** [comment]
Did you uninstall it, verify that the service file was removed and install it again verifying the apt output? Possible explanations that come to my mind:
1 you have another copy of the service file with hardcoded pi user
2 you have a pi user afterall. Run the following command: `who | grep -o -m1 '^\w*\b'` or `sudo who | grep -o -m1 '^\w*\b'` (the user which you use to install deb should be used)  It should return some user.

**Comment by D3ltorohd** [comment]
Its only one user, and no pi user. I try it again. All hyperhdr services should not there ? on systemd/system and multi.target user ?

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/514a9d46-a9b0-406d-93f2-89d3e86a5385)

Here is only one service left.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/4341bd52-6d55-44a8-89e6-71b1a00aec60)

hyperhdr@root

**Reply by awawa-dev** [reply]
`sudo systemctl disable hyperhdr@root` should remove it. This is not the standard configuration.

**Reply by awawa-dev** [reply]
if systemctl command fails, remove this file manually

**Comment by D3ltorohd** [comment]
After install hyper was there, after restart he did not start.

![image](https://github.com/awawa-dev/HyperHDR/assets/2923944/960b46fe-e225-4160-9704-8c8399e7fdf2)

**Reply by awawa-dev** [reply]
Did you remove `hyperhdr@root.service` before installing HyperHDR? This is a custom file that was not created by HyperHDR.
In your services directory you should only have the file hyperhdr.service for HyperHDR.

**Comment by D3ltorohd** [comment]
So after removing the root service, it starts after restart. Thank you very much for your help !!!
